yarn-project/txe/src/util/txe_contract_data_provider.ts

```diff
@@ -0,0 +1,30 @@
+import { type ContractArtifact, Fr } from '@aztec/aztec.js';
+import { ContractDataProvider } from '@aztec/pxe/server';
+
+export type ContractArtifactWithHash = ContractArtifact & { artifactHash: Fr };
+
+export class TXEContractDataProvider extends ContractDataProvider {
```

Comment:
Please add here a comment on why do we have this modified contract data provider.

Why isn't this optimization desired in `ContractDataProvider`?

Reply:
TBH, I don't think any of this is worth commenting. We're throwing most of this away and probably baking in some of these optimizations into PXE itself, this is all temporary.